cupy/random/distributions.py

```diff
@@ -363,6 +363,64 @@ def pareto(a, size=None, dtype=float):
     return x
 
 
+def noncentral_chisquare(df, nonc, size=None, dtype=float):
+    """Noncentral chisquare distribution.
+
+    Returns an array of samples drawn from the noncentral chisquare
+    distribution. Its probability density function is defined as
+
+    .. math::
+       f(x) = \\frac{1}{2}e^{-(x+\\lambda)/2} \\
+        \\left(\\frac{x}{\\lambda}\\right)^{k/4 - 1/2} \\
+        I_{k/2 - 1}(\\sqrt{\\lambda x})
```

Comment:
It could be mentioned that `I` is the modified Bessel function of the first kind.